scripts/game_logic/grid.c

```diff
@@ -0,0 +1,123 @@
+/*
+ ============================================================================
+ Name        : grid_gen.c
+ Author      : Liam Tapper
+ Version     :
+ Copyright   :
+ Description : General Idea for grid generation and traversal while being O(1)
+         or as close as possible to it.
+         This just holds basic game logic
+ ============================================================================
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
```

Comment:
You can guard stdio/stdlib using a #define, explained a bit more below in another comment, if you have any questions can add here.